src/hotspot/share/gc/shenandoah/shenandoah_globals.hpp

```diff
@@ -35,6 +35,28 @@
                             range,                                          \
                             constraint)                                     \
                                                                             \
+  product(bool, GenShenCensusAtEvac, true, EXPERIMENTAL,                    \
+          "Object age census at evacuation, not mearking")                  \
+                                                                            \
+  product(bool, GenShenAdaptiveTenuring, false, EXPERIMENTAL,               \
+          "Adaptive tenuring threshold")                                    \
+                                                                            \
+  product(uintx, GenShenMinTenuringThreshold, 1, EXPERIMENTAL,              \
+          "Floor for adaptive tenuring threshold")                          \
+                                                                            \
+  product(uintx, GenShenMaxTenuringThreshold, 16, EXPERIMENTAL,             \
+          "Ceiling for adaptive tenuring threshold")                        \
+                                                                            \
+  product(double, GenShenTenuringMortalityRateThreshold, 0.3, EXPERIMENTAL, \
```

Comment:
Experiment with different values here (0.1 may be more appropriate perhaps?).

Add guidance about which way (lower) to move this parm to keep objects in young gen longer.

Reply:
Done; changed down to 0.1, added documentation guidance. Experiments pending to establish what might work well most of the time.